### PR TITLE
prepare release v0.5.0

### PR DESCRIPTION
**Pull request by DirkEilander**
- bumped version to v0.5.0
- matched dependencies in pyproj.toml with all environment yml files

### CODE REVIEW

File: binder/environment.yml [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
   - requests
+  - rioxarray
   - scipy
+  - toml
   - xarray
-  - zarr
\ No newline at end of file
+  - zarr
+  - pip:
+    - hydroengine
```
Comment (hboisgon) [inline binder/environment.yml:39]: I see you added hydroengine, do we need it in hydromt core? I thought just hydromt_wflow

File: binder/environment.yml [hunk 2]
```diff
@@ -1,33 +1,40 @@
...
   - affine
-  - click
   - black
   - bottleneck
   - cartopy # to run examples
+  - click
   - dask
   - descartes # to run examples
   - entrypoints
   - flit>=3.2
```
Comment (hboisgon) [inline binder/environment.yml:16]: Shall we do flit 3.4 to be sure pip install works ?

**Review by hboisgon — APPROVED** [review]
Looks good! Just two small comments for flit minimum version and hydroengine dependency.

File: binder/environment.yml (hunk 2 above)
Comment (DirkEilander) [reply binder/environment.yml]: Good one. I've changed this in all environment files and the pyproj.toml

File: binder/environment.yml (hunk 1 above)
Comment (DirkEilander) [reply binder/environment.yml]: Removed hydroengine here and marked it optional in hydromt-dev.

**Pull request merged**